pyprof/prof/reduction.py

```diff
@@ -44,44 +42,33 @@ def __init__(self, d):
 
         # The input can be a scalar or a tensor
         if 'shape' in i:  # tensor
-            self.shape = i['shape']
-            self.type = i['dtype']
+            self.inp = Tensor(i['shape'], i['dtype'])
         else:  # scalar
             assert ('value' in i)
-            self.shape = (1, )
-            self.type = i['type']
+            self.inp = Tensor([], i['type'])
 
         self.dir = d.dir
         self.sub = d.sub
 
     def params(self):
-        p = OrderedDict([('T', self.shape), ('type', self.type)])
-        return p
+        return str(self.inp)
 
     def tc(self):
         return "-"
 
     def op(self):
-        return self.op_
+        return self._op
 
     def mod(self):
-        return self.mod_
-
-    def elems(self):
-        return Utility.numElems(self.shape)
+        return self._mod
 
     def bytes(self):
-        if self.sub == 0:
-            return self.elems() * Utility.typeToBytes(self.type)
-        else:
-            return 0
+        b = self.inp.bytes + self.inp.itemsize
```

Comment:
Not sure if this is much of an optimization. You are now always doing the add/math/store and the conditional, where as before, the math only occurred on the conditional.

This method may also obfuscate the intention as well.

Reply:
Done. Reverted.